### ISSUE TITLE
Have a look at simplification

### ISSUE THREAD

**Issue by xivk** [itinero/routing#120]
Have a look at simplification and coordinate accuracy, routes like this are bad:

http://geojson.io/#id=gist:anonymous/fdb37c00dc126946a0d5922b32ae48bb&map=19/1.30078/103.84289

**Comment by xivk** [comment]
This is part of a larger issue, the precision of the coordinates in general is bad, this will lead to a breaking change.

Postponed to Itinero v2.

**Comment by ironromeo** [comment]
Please explain "the precision of the coordinates in general is bad". This can be very important taking into account that I use "Coordinate" in my code.

**Comment by xivk** [comment]
Nothing to worry about for now, but using floats in the coordinate class could lead to inaccurate calculations, at least not accurate enough for everywhere in world. If you look at this route:

http://geojson.io/#id=gist:anonymous/fdb37c00dc126946a0d5922b32ae48bb&map=19/1.30078/103.84289

The artifacts you are seeing is the effect I'm talking about. This is a pretty big breaking change so I won't be changing this until Itinero v2. It also means changing the way coordinates are stored in the RouterDb.

In short, if everything has been working fine up until now for you, it will continue to do so.

**Comment by xivk** [comment]
Thought about this some more:

- [ ] Recheck the cause of this issue.

**Comment by xivk** [comment]
Closing this, it's part of the Itinero 2.0 roadmap:

http://docs.itinero.tech/docs/itinero/development/index.html
